### PR TITLE
Added support for DaemonSet for akeyless gateway

### PR DESCRIPTION
**Pull request by OriBenHur-akeyless**
This pr aims to add the ability to deploy Akeyless Gateway as a DaemonSet instead of as a deployment

### CODE REVIEW

**Review by ak-nir — APPROVED** [review]

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 apiVersion: apps/v1
+{{- if eq .Values.kind "Deployment" }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:2]: Deployment should be default value and change to DaemonSet only if the user asked for it

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 2]
```diff
@@ -13,8 +19,7 @@ metadata:
   {{- toYaml .Values.deployment.annotations | nindent 4 }}
   {{- end }}
 spec:
-
-  {{- if not .Values.HPA.enabled }}
+  {{- if and (eq .Values.kind "Deployment") (not .Values.HPA.enabled) }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:22]: what with HPA yaml?
If set to DaemonSet the HPA should not be applied on YAML level as well

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 1 above)
Comment (OriBenHur-akeyless) [reply charts/akeyless-api-gateway/templates/deployment.yaml:2]: It's the default see the else at the end of the condition

File: charts/akeyless-api-gateway/templates/deployment.yaml (hunk 2 above)
Comment (OriBenHur-akeyless) [reply charts/akeyless-api-gateway/templates/deployment.yaml:22]: You are right i'll fix it

**Review by niv-ez — APPROVED** [review]

**Pull request merged**